unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/WebRequest/WebRequestController.cs

```diff
@@ -154,14 +163,36 @@ public static WebRequestController Create()
             int requestAttemps,
             int timeout,
             CancellationToken cancellationToken,
-            Dictionary<string, string> headers = null) where T : IWebRequestFactory
+            Dictionary<string, string> headers = null,
+            bool isSigned = false) where T : IWebRequestFactory
         {
             requestAttemps = Mathf.Max(1, requestAttemps);
+
             for (int i = requestAttemps - 1; i >= 0; i--)
             {
                 UnityWebRequest request = requestFactory.CreateWebRequest(url);
                 request.timeout = timeout;
 
+                if (!Enum.TryParse(request.method, out RequestMethod method))
+                    method = RequestMethod.Get;
+
+                if (isSigned && rpcSignRequest != null)
```

Comment:
Is it valid for `rpcSignRequest` to be `null` in any scenario? if not, I'd let it crash or `Assert` it in the constructor.